setup.py

```diff
@@ -8,9 +9,26 @@
 def read(fname):
     return open(os.path.join(os.path.dirname(__file__), fname)).read()
 
+def extract_version():
+    with open('version.h') as inf:
+        lines = inf.readlines()
+
+    ver_map = {'MAJOR': 0,
```

Comment:
Ahh... we do control this object.  Can you just make a simple version object out of it with a cleaner `to_str` method